user-registration-app/src/app/views/registration-form/registration-form.component.ts

```diff
@@ -128,108 +89,87 @@ export class RegistrationFormComponent implements OnInit {
   }
 
   private consolidateEthnicities(): string {
-    let consolidatedEthnicity = "";
-
-    // loop through the ethnicities object; append sel ethnicities to the ethnicity string
-    for (let selEthnicity in this.registrationForm.ethnicities) {
-      if (this.registrationForm.ethnicities[selEthnicity]) {
-        consolidatedEthnicity += `${selEthnicity}, `;
-      }
-    }
-
-    // remove the trailing comma before returning
-    if (consolidatedEthnicity.slice(-2) === ", ") {
-      consolidatedEthnicity = consolidatedEthnicity.slice(0, -2);
-    }
-
-    return consolidatedEthnicity;
+    const possibleEthnicities = Object.keys(this.registrationForm.ethnicities);
+    const selectedEthnicities = possibleEthnicities.filter(ethnicity => this.registrationForm.ethnicities[ethnicity]);
+    return selectedEthnicities.join(', ');
   }
 
   private consolidateAddress() {
-    let consolidatedAddress = "";
-
-    for (let addrField in this.registrationForm.addressFields) {
-      if (this.registrationForm.addressFields[addrField]) {
-        consolidatedAddress += `${this.registrationForm.addressFields[addrField]}, `;
-      }
-    }
-
-    if (consolidatedAddress.slice(-2) === ", ") {
-      consolidatedAddress = consolidatedAddress.slice(0, -2);
-    }
-
-    this.registrationForm.address = consolidatedAddress;
-  }
-
-  sanitizeUrl(resume_link: any) {
-    if (!(resume_link instanceof URL)) {
-      throw new Error('Must be a URL');
-    }
-    return resume_link.href;
+    return Object.values(this.registrationForm.addressFields).filter(field => field).join(', ');
   }
 
   private validate() {
     const result = this.validator(this.registrationForm);
     if (!result) {
-      this.validator.errors
-        .map(error => this.showError(
-          RegistrationFormComponent.getFormattedErrorText(error),
-          2),
-        );
+      this.validator.errors.map(
+        error => this.toastrService.warning(RegistrationFormComponent.getFormattedErrorText(error)),
+      )
     }
     return result;
   }
 
   constructor(public router: Router,
               private route: ActivatedRoute,
               private httpService: HttpService,
-              private alertsService: AlertService,
+              private toastrService: ToastrService,
               private authService: AuthService) {
-    console.log(data);
     this.registrationForm = new Registration();
     RegistrationFormComponent.regFormComp = this;
     this.prettifiedPhone = '';
     this.asYouType = new AsYouType('US');
-    this.validator = ajv.compile(registeredUserSchema);
+    this.validator = ajv.compile(registeredUserSchema.default);
   }
 
   ngOnInit() {
     this.route.data.subscribe(({ registration }) => {
       this.registrationForm = new Registration();
 
-      //Data to keep
-      this.nukeData(this.registrationForm, registration)
-      
+      // Data to keep
```

Comment:
how dare you delete my perfectly named function nukeData in which data was not nuked >:( it didn't create any confusion at all

Reply:
My pacifist tendencies overtook my love for poorly named functions. Not much I could do about it :shrug: